exp/agents/mrkl/mrkl.go

```diff
@@ -0,0 +1,167 @@
+package mrkl
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tmc/langchaingo/exp/agents/agentExecutor"
+	"github.com/tmc/langchaingo/exp/chains"
+	"github.com/tmc/langchaingo/exp/tools"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/schema"
+)
+
+type OneShotZeroAgent struct {
+	llm        llms.LLM
+	query      string
+	chain      chains.Chain
+	tools      []tools.Tool
+	verbose    bool
+	maxRetries int
+}
+
+type OneShotZeroAgentOptions map[string]any
+
+var _ agentExecutor.AgentExecutor = (*OneShotZeroAgent)(nil)
+
+const FinalAnswerAction = "Final Answer:"
+
+func checkOptions(opts OneShotZeroAgentOptions) OneShotZeroAgentOptions {
+	if _, ok := opts["verbose"].(bool); !ok {
+		opts["verbose"] = false
+	}
+	if _, ok := opts["maxRetries"].(int); !ok {
+		opts["maxRetries"] = 3
+	}
+	return opts
+}
+
+//Aqui se crea el chain
```

Comment:
please put comments into the form of " NewShotAgent ..." and have the comments be in english.

Reply:
Sorryy hahhaa , forgot to delete that ahahah